src/file.cpp

```diff
@@ -1330,6 +1351,7 @@ namespace libtorrent
 	bool file::open(std::string const& path, int mode, error_code& ec)
 	{
 		close();
+		native_path_string const &file_path = convert_to_native_path_string(path);
```

Comment:
I see, it doesn't return a reference. you're relying on the fact that holding a reference to an object extends its lifetime. I don't think there's any benefit of doing this over just storing this directly in a local variable. is there?
I think it would be a lot clearer, in all cases where you call this function, to store it in a string by value.